homeassistant/components/camect/camera.py

```diff
@@ -0,0 +1,204 @@
+from aiohttp import web
+import async_timeout
+
+from homeassistant.components import camect, camera
+from homeassistant.components.http import HomeAssistantView
+
+REQUIREMENTS = ['camect-py==0.1.0']
+DEPENDENCIES = ['camect']
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, cam_ids):
+    """Add an entity for every camera from Camect Home."""
+    component = hass.data[camera.DOMAIN]
+    hass.http.register_view(CamectWebsocketView(component))
+    hass.http.register_view(CamectBundleJsView())
+    hass.http.register_view(CamectFontFileView())
+
+    home = hass.data[camect.DOMAIN]
+    camect_site = home.get_cloud_url('')
+    cam_jsons = home.list_cameras()
+    if cam_jsons:
+        cams = []
+        for cj in cam_jsons:
+            if not cam_ids or cj['id'] in cam_ids:
+                cams.append(Camera(cj, camect_site))
+        add_entities(cams, True)
+    return True
+
+
+class Camera(camera.Camera):
+    """An implementation of a camera supported by Camect Home."""
+
+    def __init__(self, json: Dict[str, str], camect_site: str):
+        """Initialize a camera supported by Camect Home."""
+        super(Camera, self).__init__()
+        self._device_id = json['id']
+        self._id = '{}_{}'.format(camect.DOMAIN, self._device_id)
+        self.entity_id = '{}.{}'.format(camect.DOMAIN, self._id)
+        self._name = json['name']
+        self._make = json['make'] or ''
+        self._model = json['model'] or ''
+        self._url = json['url']
+        self._width = int(json['width'])
+        self._height = int(json['height'])
+        self._camect_site = camect_site
+
+    @property
+    def name(self):
+        """Return the name of this camera."""
+        return self._name
+
+    @property
+    def brand(self):
+        """Return the camera brand."""
+        return self._make
+
+    @property
+    def model(self):
+        """Return the camera model."""
+        return self._model
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return True
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._id
+
+    @property
+    def entity_picture(self):
+        """Return a link to the camera feed as entity picture."""
+        return None
+
+    def camera_image(self):
+        return None
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            'device_id': self._device_id,
+            'device_url': self._url,
+            'video_width': self._width,
+            'video_height': self._height,
+            'camect_site': self._camect_site,
+        }
+
+    @property
+    def should_poll(self):
+        """No need for the poll."""
+        return False
+
+
+class CamectWebsocketView(camera.CameraView):
+    """Camect view to proxy Websocket to home."""
+
+    url = '/api/camect_proxy/websocket/{entity_id}'
```

Comment:
This does not belong in Home Assistant.

Reply:
removed. These were used to support a new card.